### PR TITLE
MFB-1189 is_dependent() has no relationship test — unrelated members wrongly enter the tax unit (EITC understated)

### PR DESCRIPTION
**Pull request by IrynaKolh**
## Context & Motivation

- Fixes #[MFB-1189](https://linear.app/myfriendben/issue/MFB-1189/is-dependent-has-no-relationship-test-unrelated-members-wrongly-enter)

## Changes Made
- Added relationship gating to [HouseholdMember.is_dependent()](vscode-file://vscode-app/c:/Users/irica/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-browser/workbench/workbench.html) in [models.py](vscode-file://vscode-app/c:/Users/irica/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-browser/workbench/workbench.html) for both:
* qualifying-child path
* qualifying-relative path
- Introduced shared constant [DEPENDENT_ELIGIBLE_RELATIONSHIPS](vscode-file://vscode-app/c:/Users/irica/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-browser/workbench/workbench.html) to avoid hardcoded branch-specific relationship checks.
- Explicitly excluded unrelated relationships ([roommate](vscode-file://vscode-app/c:/Users/irica/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-browser/workbench/workbench.html), boyfriendOrGirlfriend) from dependent eligibility.
- Updated dependent relationship set to align with IRS notes (including removing fosterParent from eligible set).
- Added/expanded tests in [test_is_dependent.py](vscode-file://vscode-app/c:/Users/irica/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-browser/workbench/workbench.html):
* relationship coverage across both paths
* student roommate regression (Path 1)
* low-income unrelated housemate regression (Path 2)
* domestic partner behavior coverage
- Added tax-unit regression tests in [test_qualifying_relative_units.py](vscode-file://vscode-app/c:/Users/irica/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-browser/workbench/workbench.html) to ensure unrelated members stay out of [main_tax_unit](vscode-file://vscode-app/c:/Users/irica/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-browser/workbench/workbench.html).
_ Updated relatedOther label text in [base.py](vscode-file://vscode-app/c:/Users/irica/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-browser/workbench/workbench.html) for clearer UX intent:
"Other relative (aunt, uncle, cousin, in-law, etc.)"

## Testing

- Migrations to run: no
- Configuration updates needed: yes
- Environment variables/settings to add: no
- Manual testing steps:

  1. Create a household with:
     - Head of household (`headOfHousehold`)
     - Optional spouse/domestic partner
     - Additional member with varying `relationship`, age/student/disability, and income.
  2. Verify `is_dependent()` behavior for both branches:
     - **Path 1 (qualifying child proxy)**: age/student/disability + income ratio
     - **Path 2 (qualifying relative proxy)**: below IRS threshold
  3. Confirm relationship gate is applied to **both** paths:
     - `roommate` and `boyfriendOrGirlfriend` should **not** be treated as dependents even if they satisfy income/age conditions.
  4. Generate PolicyEngine input and verify tax unit membership:
     - Unrelated members should be in `secondary_tax_unit` (not `main_tax_unit`).
  5. Confirm no regressions:
     - Legitimate relationships (`child`, `parent`, `sisterOrBrother`, `grandChild`, `relatedOther`, etc.) still behave as expected based on income/age criteria.

#### Scenario 1: Two Student Roommates (Path 1 — qualifying child gate)
Unrelated student roommate must NOT be treated as head's dependent.
Screener input:
Field | Member 1 (Head) | Member 2 (Roommate)
-- | -- | --
Household Size | 2 |  
Relationship | (head, auto) | Roommate
Age | 30 | 22
Student | No | Yes (full-time)
Income | Wages, $20,000/yearly | Wages, $3,000/yearly

Scenario 1b (control — same but with "Child" relationship): Change Member 2's relationship from "Roommate" to "Child". Everything else identical. The child SHOULD be in the head's tax unit as a qualifying child. Compare EITC results — they should differ if the gate is working.
image

image

---

#### Scenario 2 – Low-Income Partner (Qualifying Relative)
Screener input:
Field | Member 1 (Head) | Member 2 (Partner)
-- | -- | --
Household Size | 2 |  
Relationship | (head, auto) | Boyfriend/Girlfriend
Age | 35 | 30
Student | No | No
Income | Wages, $25,000/yearly | Wages, $1,000/yearly

image

#### Scenario 2b – Control
Change relationship to "Child" (age 30, $1k income). Now they SHOULD qualify as a qualifying relative and enter the head's tax unit. Results should differ.
Screenshot 2026-07-23 132110

---

#### Scenario 3 – Higher-Income Roommate (Regression)
- **Relationship:** Roommate
- **Income:** $18,000
- **Expected:** Excluded from the head's tax unit. EITC calculated correctly.

Screener input:
Field | Member 1 (Head) | Member 2 (Child) | Member 3 (Roommate)
-- | -- | -- | --
Household Size | 3 |   |  
Relationship | (head, auto) | Child | Roommate
Age | 35 | 8 | 22
Student | No | No | Yes
Income | Wages, $20,000/yearly | None | Wages, $18,000/yearly

image

without roommate 

image

---

#### Scenario 4 – Legitimate Dependents
Screener input:
Field | Head | Member 2 | Member 3
-- | -- | -- | --
Household Size | 3 |   |  
Relationship | (head) | Child | Parent
Age | 40 | 8 | 70
Student | No | No | No
Income | Wages, $40,000/yr | None | Social Security, $4,000/yr
image

Both child and parent should be in the head's tax unit. 
Child qualifies via Path 1 (age ≤ 18).
Parent qualifies via Path 2 (income $4k &lt; ~$5,300 threshold). 
EITC/CTC should reflect all three in the same tax unit

if change "Parent" to "Other relative (aunt, uncle, cousin, in-law, etc.)"
image

---

#### Scenario 5 – Domestic Partner
Purpose: Domestic partner should pair with head as spouse, not enter dependent logic
Field | Head | Member 2
-- | -- | --
Household Size | 2 |  
Relationship | (head) | Domestic Partner
Age | 35 | 33
Income | Wages, $30,000/yr | None ($0)

image

Despite $0 income (which would easily pass the qualifying-relative threshold), the domestic partner should NOT be listed as a dependent
They should be treated as a spouse — results should show filing status as married/joint equivalent
Compare with a Scenario where Member 2 is "Spouse" — results should be identical

---

#### Scenario 6 – relatedOther Label (FE)
- **Expected:** Updated label displays **"Other relative (aunt, uncle, cousin, in-law, etc.)"**.

## Deployment

- Post-deployment scripts needed: no
- Production config updates: no
- Admin updates needed: yes,  `add_translations`
- Notify team/users of:  Superadmin access is needed to manually update this translation during deployment, since a saved value for relationshipOptions.relatedOther likely already exists in staging/production and may override the new code default.

## Notes for Reviewers

- Known limitations:
- Future considerations:

## Summary by CodeRabbit

- **New Features**
  - Dependent eligibility now considers the household member’s relationship type, improving classification for qualifying children and relatives.
  - Unrelated roommates and housemates are correctly excluded from the primary tax unit when appropriate.

- **Updates**
  - Clarified the “Other relative” relationship option with examples such as aunt, uncle, cousin, and in-law.

### CODE REVIEW

File: screener/models.py [hunk 1]
```diff
@@ -14,6 +14,31 @@
...
+        "fosterChild",
+        "stepChild",
+        "grandChild",
+        "parent",
+        "stepParent",
+        "grandParent",
+        "sisterOrBrother",
+        "stepSisterOrBrother",
+        "relatedOther",
+    }
```
[the remark this answers is not in the document]
Comment (IrynaKolh) [reply screener/models.py:37]: The comment is technically correct per IRS rules — parents/grandparents/stepParents are ancestors and can never be qualifying children (only descendants and siblings can). However, the practical impact is negligible:

* Minor parent (age ≤ 18): Impossible — they'd be younger than their own child who is the head of household.
* Student parent (age ≤ 23): The head's parent being 23 or younger is essentially impossible.
* Disabled parent: The only realistic case, but a disabled parent with income below the ~$5,300 threshold would pass Path 2 (qualifying relative) anyway — same end result. It only diverges for a disabled parent with income between $5,300 and 50% of household income, which is a very narrow edge case.

The relatedOther/cousin point is an intentional, documented tradeoff — we don't collect the full-year residency or >50% support data required for the IRS non-relative household member test (noted as out of scope in the ticket).

Deferring the per-path split as a future refinement since it doesn't affect any realistic household scenario.

File: configuration/white_labels/base.py [hunk 2]
```diff
@@ -143,7 +143,10 @@ def get_white_label(self) -> WhiteLabel:
         },
         "domesticPartner": {"_label": "relationshipOptions.domesticPartner", "_default_message": "Domestic Partner"},
         "roommate": {"_label": "relationshipOptions.roommate", "_default_message": "Roommate"},
-        "relatedOther": {"_label": "relationshipOptions.relatedOther", "_default_message": "Related in some other way"},
+        "relatedOther": {
+            "_label": "relationshipOptions.relatedOther",
+            "_default_message": "Other relative (aunt, uncle, cousin, in-law, etc.)",
+        },
     }
```
Comment (nihalerdal) [inline configuration/white_labels/base.py:150]: Nice catch on the rename Iryna that fixes exactly what the ticket calls out.

Just a suggestion: maybe mark "Admin updates needed" as yes. Otherwise the Translation database row won't update automatically, and users will still see the old text.

**Review by nihalerdal — APPROVED** [review]
LGTM! Approving with a small non-blocking suggestion (translation admin step)

**Comment by Linda-Vu** [comment]
@IrynaKolh could you update the deployment section of your PR to add a note that says something like:

"Note: Superadmin access is needed to manually update this translation during deployment, since a saved value for relationshipOptions.relatedOther likely already exists in staging/production and may override the new code default."

**Review by andreaHG — APPROVED** [review]
